lib/candy_check/app_store/unified/in_app_receipt.rb

```diff
@@ -0,0 +1,94 @@
+module CandyCheck
+  module AppStore
+    module Unified
+      # Describes a receipt for an in-app purchase.
+      # It's part of ios 7 style unified receip format
+      class InAppReceipt
+        include Utils::AttributeReader
+        # @return [Hash] the raw attributes returned from the server
+        attr_reader :attributes
+
+        # Initializes a new instance which bases on a JSON result
+        # from Apple's verification server
+        # @param attributes [Hash]
+        def initialize(attributes)
+          @attributes = attributes
+        end
+
+        # The number of items purchased
+        # @return [Integer]
+        def quantity
+          read_integer('quantity')
+        end
+
+        # The product identifier of the item that was purchased
+        # @return [String]
+        def product_id
+          read('product_id')
+        end
+
+        # The transaction identifier of the item that was purchased
+        # @return [String]
+        def transaction_id
+          read('transaction_id')
+        end
+
+        # For a transaction that restores a previous transaction,
+        # the transaction identifier of the original transaction.
+        # Otherwise, identical to the transaction identifier
+        # @return [String]
+        def original_transaction_id
+          read('original_transaction_id')
+        end
+
+        # The date and time that the item was purchased
+        # @return [DateTime]
+        def purchase_date
+          read_datetime_from_string('purchase_date')
+        end
+
+        # For a transaction that restores a previous transaction,
+        # the date of the original transaction
+        # @return [DateTime]
+        def original_purchase_date
+          read_datetime_from_string('original_purchase_date')
+        end
+
+        # For a transaction that was canceled by Apple customer support,
+        # the time and date of the cancellation
+        # @return [DateTime]
+        def cancellation_date
+          read_datetime_from_string('cancellation_date')
+        end
+
+        CANCELATION_REASONS = {
+          1 => 'Customer canceled their transaction due to an actual' \
+               'or perceived issue within your app.',
+          0 => 'Transaction was canceled for another reason, for example, if' \
+                'the customer made the purchase accidentally.'
+        }.freeze
+
+        # For a transaction that was cancelled, the reason for cancellation
+        # @return [String]
+        def cancellation_reason
```

Comment:
Maybe it would make sense to keep the model as close to the response from Apple as possible?

Perhaps 
```
def cancellation_reason
  read_integer('cancellation_reason')
end
``` 
and
``` 
def cancellation_reason_string
  code = read_integer('cancellation_reason')
  code && CANCELATION_REASONS[code] 
end
```

Reply:
And perhaps the same approach for 
```
        EXPIRATION_INTENTS = {
          1 => 'Customer canceled their subscription.',
          2 => 'Billing error; for example customer’s payment information'\
               'was no longer valid.',
          3 => 'Customer did not agree to a recent price increase.',
          4 => 'Product was not available for purchase at the time of renewal.',
          5 => 'Unknown error.'
        }.freeze

        # For an expired subscription, the reason for the
        # subscription expiration.
        # @return [Integer]
        def expiration_intent
          read_integer('expiration_intent')
        end

        # For an expired subscription, the reason for the
        # subscription expiration.
        # @return [String]
        def expiration_intent_string
          code = expiration_intent
          code && EXPIRATION_INTENTS[code]
        end
```

Reply:
I like this approach. I'm now back to the office and will start catching up all the issues in the next days and weeks.